src/plugins/intel_cpu/src/cpu_streams_calculation.cpp

```diff
@@ -0,0 +1,166 @@
+                                             : std::min(proc_type_table[0][MAIN_CORE_PROC], input_threads);
+        }
+
+        if (0 != input_streams) {
+            if (input_streams >= n_threads) {
+                n_streams = n_threads;
+                n_threads_per_stream = 1;
+            } else {
+                n_streams = input_streams;
+                n_threads_per_stream =
+                    std::min(std::max(1, n_threads / input_streams), proc_type_table[0][MAIN_CORE_PROC]);
+                if (proc_type_table.size() == 1) {
+                    if ((n_threads_per_stream > proc_type_table[0][MAIN_CORE_PROC]) &&
+                        (n_threads_per_stream < proc_type_table[0][MAIN_CORE_PROC] * 2)) {
+                        n_threads_per_stream = proc_type_table[0][MAIN_CORE_PROC];
+                    } else if (n_threads_per_stream < proc_type_table[0][MAIN_CORE_PROC]) {
+                        n_threads_per_stream = int(
+                            proc_type_table[0][MAIN_CORE_PROC] /
+                            ((proc_type_table[0][MAIN_CORE_PROC] + n_threads_per_stream - 1) / n_threads_per_stream));
+                    }
+                }
+            }
+        } else {
+            if (0 == model_prefer_threads) {
+                int n_proc = std::min(n_threads, proc_type_table[0][MAIN_CORE_PROC]);
+                if (0 == n_proc % 4) {
+                    n_threads_per_stream = 4;
+                } else if (0 == n_proc % 5) {
+                    n_threads_per_stream = 5;
+                } else if (0 == n_proc % 3) {
+                    n_threads_per_stream = 3;
+                } else if (proc_type_table.size() == 1) {
+                    n_threads_per_stream = n_proc;
+                } else {
+                    n_threads_per_stream = (n_proc > 16) ? 4 : std::max(1, static_cast<int>(n_proc / 4));
+                }
+                n_streams = static_cast<int>(n_threads / n_threads_per_stream);
+
+                while (n_streams < n_threads_per_stream) {
+                    if (1 == n_threads_per_stream) {
+                        break;
+                    } else {
+                        n_threads_per_stream = static_cast<int>((n_threads_per_stream * 2 - 1) / 2);
+                        n_threads_per_stream = static_cast<int>(
+                            proc_type_table[0][MAIN_CORE_PROC] /
+                            ((proc_type_table[0][MAIN_CORE_PROC] + n_threads_per_stream - 1) / n_threads_per_stream));
+                        n_streams = static_cast<int>(n_threads / n_threads_per_stream);
+                    }
+                }
+            } else {
+                n_streams = ((n_threads + model_prefer_threads - 1) / model_prefer_threads);
+                n_threads_per_stream = static_cast<int>(n_threads / n_streams);
+            }
+        }
+
+        stream_info[THREADS_PER_STREAM] = n_threads_per_stream;
+
+        if (proc_type_table.size() == 1) {
+            while (1) {
```

Comment:
Is there any risk enter dead-loop? such as in some corner case stream_info[THREADS_PER_STREAM] = 0